GSREN3D-133: Update style for network figure item

ismailsunni opened the pull request:
<!-- Title must be: GSREN3D-XXX: Description of changes -->

### JIRA issue

https://jira.camptocamp.com/browse/GSREN3D-133

### Description

- We can reuse the network figure component, only needs to add v-if for the icon and more information icon
- Adapt the new style from the latest Figma design
- Add more variants in the histoire
- Add the missing icons

### Screenshots

![image](https://user-images.githubusercontent.com/1421861/197905778-3332367b-ec73-4b23-aa2d-ebdc5b97cc74.png)

fgravin commented on src/components/ui/UiNetworkFigure.vue:
```diff
@@ -1,20 +1,31 @@
-      <span class="text-base px-1">{{ props.unit }}</span>
+  <div class="w-32 h-24 font-dm-sans">
+    <div class="flex flex-row items-center p-0 gap-2 w-32 h-9">
+      <div class="flex flex-row leading-8 font-bold gap-1 items-end flex-grow">
```
where did you find the class `flex-grow` ?
https://tailwindcss.com/docs/flex-grow

BTW in figma it's `grow:0` not `grow:1`

fgravin commented on src/components/ui/UiNetworkFigure.vue:
```diff
@@ -1,20 +1,31 @@
-      <span class="text-4xl px-1">{{ props.figure }}</span>
-      <span class="text-base px-1">{{ props.unit }}</span>
+  <div class="w-32 h-24 font-dm-sans">
+    <div class="flex flex-row items-center p-0 gap-2 w-32 h-9">
```
`flex-row` is useless as it's the default

fgravin commented on src/components/ui/UiNetworkFigure.vue:
```diff
@@ -1,20 +1,31 @@
-    >
-      {{ props.description }}
+    <div class="flex flex-row items-center p-0 gap-2 w-28 h-10">
+      <div class="font-normal text-sm leading-5 text-black flex-grow">
```
black should be the main text color

fgravin reviewed (changes requested):
Good work but there are some layout issues
![Screenshot from 2022-10-26 15-39-24](https://user-images.githubusercontent.com/1491924/198041925-165a54e7-2cc7-4eda-949f-f156e375d3c3.png)

Globally, try to avoid setting size on the elements, it is in figma but it does not mean we need it in the CSS. Usually, the sizing is done automatically regarding the content, the layout etc...

plaulhe commented:
Are the accessibility recommendation  useless? 
i don't see the aria-labelledby attribute. 
Does the screen reader works well?

ismailsunni commented:
@fgravin I have addressed your review. No more setting size on the component, but the setting should be set when using the component. Here are the new looks:
![image](https://user-images.githubusercontent.com/1421861/199408029-75905454-1f6c-4fd6-b447-ff58126ddcd2.png)

@plaulhe I have checked with the screen reader, and it seems to work well.

fgravin commented on src/components/line/LineFigures.vue:
```diff
@@ -0,0 +1,48 @@
+  <div
+    class="flex flex-row px-4 py-3 gap-3 bg-slate-50 border rounded-lg border-slate-100"
+  >
+    <template
```
why do you use a extra `template` here ?

fgravin replied:
I think that each block should be an `<article>`
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article

ismailsunni replied:
@fgravin because of this https://vuejs.org/guide/essentials/list.html#v-for-with-v-if I need to use a template so that I can use v-if on v-for.

fgravin reviewed (commented):
I don't have time now to test or check that in depth. I've just made 2 comments.
Thanks for the work

plaulhe commented:
Ismail, that's a great work. 
i used the screen reader and it's almost good. 
But, we have to find a way separate the icon from the read. 
I mean: 128 km Nouvelles lignes must be read straight and then the Icon. 
More over, the icon is read like : lines.5232b718.svg
We also have to find a way to read : ' new lines icon'

ismailsunni commented:
@plaulhe which screen reader are you using? I use the Google Chrome screen reader extension. It seems good.
The icon is read "Line Image" by the screen reader. Perhaps different screen readers read differently. I tried the one from Ubuntu, and it's not as good as the one from Google Chrome.

akhelouat commented on src/components/home/NetworkFigures.vue:
```diff
@@ -12,17 +13,39 @@ const state = reactive({
+function getLength(networkFigures: null | NetworkFigureModel[]): Number {
+  if (networkFigures == null) {
+    return 0
+  } else {
```
I think that there is no need for else statement here

akhelouat commented on src/components/line/LineFigures.vue:
```diff
@@ -0,0 +1,50 @@
+function getLength(networkFigures: null | LineFigureModel[]): Number {
+  if (networkFigures == null) {
+    return 0
+  } else {
```
same as the previous comment about the else statement

fgravin commented:
There was some issue with the box sizing of the figures blocks.
I tried to fit perfectly to the Figma.

fgravin reviewed (approved):
I think we can merge it, to me it looking exactly like the Figma, but needed some tricks though...
Might be enough for now.

Thanks @ismailsunni
